orchestrator-service/internal/broker/kafka.go

```diff
@@ -0,0 +1,73 @@
+package broker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/segmentio/kafka-go"
+)
+
+type Broker struct {
+	conn *kafka.Conn
+}
+
+func New(c *kafka.Conn) *Broker {
+	return &Broker{
+		conn: c,
+	}
+}
+
+type Message struct {
+	Link string `json:"link"`
+}
+
+func (b *Broker) Write(mes []byte) error {
+	_, err := b.conn.Write(mes)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *Broker) Read() (arr []Message, err error) {
+	reader := newKafkaReader("localhost:9092", "processed-images-topic")
+	defer reader.Close()
+	var message Message
+
+	for {
+		m, err := reader.ReadMessage(context.Background())
+		if err == io.EOF {
+			err = json.Unmarshal(m.Value, &message)
+			if err != nil {
+				return []Message{}, err
+			}
+
+			fmt.Println(message.Link)
+			arr = append(arr, message)
+			break
+		}
+
+		err = json.Unmarshal(m.Value, &message)
+		if err != nil {
+			return []Message{}, err
+		}
+
+		fmt.Println(message.Link)
+		arr = append(arr, message)
+	}
+
+	err = reader.Close()
```

Comment:
Add it into `defer` function...

Reply:
But it in this place we should overwrite existing EOF error, to prevent next layers from `err !=nil`

Reply:
![image](https://user-images.githubusercontent.com/70835145/220322926-5fa2f4df-85b8-409c-aace-638ca3c165d6.png)
Giving example in kafka-go docs